lib/kitchen/directions/bake_footnotes/v2.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Kitchen::Directions::BakeFootnotes
+  # Difference from v1:
+  # 1. top-level pages are numbered with roman numerals
+  # 2. footnote counts don't reset with each new chapter
+  class V2
+    def bake(book:)
+      book.body.element_children.only(Kitchen::PageElement,
+                                      Kitchen::CompositePageElement,
+                                      Kitchen::CompositeChapterElement).each do |page|
```

Comment:
You may not need `Kitchen::` on these three lines, ditto V1.  They aren't needed when used in BakeToc this way.